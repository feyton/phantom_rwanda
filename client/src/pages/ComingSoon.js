import React from 'react';
import { Link, useNavigate } from 'react-router-dom';

function ComingSoon({ title }) {
  const navigate = useNavigate();
  return (
    <div className="justify-self-center w-full h-full relative bg-gray-300 md:min-w-[300px] min-h-[300px] p-9">
      <div className="flex flex-col items-center justify-center top-0 left-0 right-0 bottom-0">
        <h2 className="text-xl font-bold text-primary  pb-5">
          You tried to access <b className="text-black">{`<<${title}>>`}</b> but
          this page is...
        </h2>
        <h1 className="text-3xl font-bold pb-5">Coming Soon</h1>
        <div className="flex flex-row w-fit justify-between">
          <Link
            to="/"
            className="bg-primary font-bold px-3 py-1 rounded-md mr-2"
          >
            Home
          </Link>
          <button
            onClick={() => navigate(-1)}
            className="bg-transparent border-2 border-primary font-bold font-rale rounded-md px-3 py-1"
          >
            {'<< Back'}
          </button>
        </div>
      </div>
    </div>
  );
}

export default ComingSoon;
