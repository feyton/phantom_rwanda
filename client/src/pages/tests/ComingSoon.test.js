import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ComingSoon from '../ComingSoon.js';

describe('Comming soon ', () => {
  it('Should render coming soon page', () => {
    const elem = renderer
      .create(
        <BrowserRouter>
          <ComingSoon />
        </BrowserRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
