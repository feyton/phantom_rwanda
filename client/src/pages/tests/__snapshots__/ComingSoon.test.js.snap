// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comming soon  Should render coming soon page 1`] = `
<div
  className="justify-self-center w-full h-full relative bg-gray-300 md:min-w-[300px] min-h-[300px] p-9"
>
  <div
    className="flex flex-col items-center justify-center top-0 left-0 right-0 bottom-0"
  >
    <h2
      className="text-xl font-bold text-primary  pb-5"
    >
      You tried to access 
      <b
        className="text-black"
      >
        &lt;&lt;undefined&gt;&gt;
      </b>
       but this page is...
    </h2>
    <h1
      className="text-3xl font-bold pb-5"
    >
      Coming Soon
    </h1>
    <div
      className="flex flex-row w-fit justify-between"
    >
      <a
        className="bg-primary font-bold px-3 py-1 rounded-md mr-2"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <button
        className="bg-transparent border-2 border-primary font-bold font-rale rounded-md px-3 py-1"
        onClick={[Function]}
      >
        &lt;&lt; Back
      </button>
    </div>
  </div>
</div>
`;
